cmd/grr/workflow.go

```diff
@@ -110,21 +120,49 @@ func diffCmd() *cli.Command {
 func applyCmd() *cli.Command {
 	cmd := &cli.Command{
 		Use:   "apply <resource-path>",
-		Short: "apply local resources to remote endpoints",
+		Short: "apply local resources to remote endpoints, -s flag is only applicable to dashboards",
 		Args:  cli.ArgsExact(1),
 	}
 	var opts grizzly.Opts
 
 	cmd.Run = func(cmd *cli.Command, args []string) error {
+		if err := checkDashboardTarget(opts); err != nil {
+			return err
+		}
+
 		resources, err := grizzly.Parse(args[0], opts)
 		if err != nil {
 			return err
 		}
 		return grizzly.Apply(resources)
 	}
+
+	cmd.Flags().StringVarP(&opts.FolderUID, "folder", "f", generalFolderUID, "folder to push dashboards to")
+	cmd = initialiseSpecOnly(cmd, &opts)
```

Comment:
Nit; Even if re-used in different commands, I'd prefer this be inlined (easier to read, less jumps), but I leave it up to you!